### ISSUE TITLE
Compiler warning in GAP 4.8.5 with gcc 6.2.0 

### ISSUE THREAD

**Issue by alex-konovalov** [gap-system/gap#924]
Reported by Bill Allombert to GAP Support: there is a warning with GAP 4r8p5 when compiled with gcc 6.2.0 (but not with 4.9.2):

```
gcc -I. -I../.. -DCONFIG_H   -Wall -g -O2  -g -O2  -o scanner.o -c
../../src/scanner.c
../../src/scanner.c: In function ‘InitKernel’:
../../src/scanner.c:3094:15: warning: array subscript is below array
bounds [-Warray-bounds]
    TLS(Input)--;
              ^
```

**Comment by dimpase** [comment]
same warning with gcc 6.2.1, by the way.

**Comment by ChrisJefferson** [comment]
This is a very, very long-standing piece of code, which is technically invalid behaviour, but in practice fine.

This basically a hack to open the first used file in `OpenInput`, by pushing a pointer to one before the start or an array, which is then checked by lines like `If (Input +1 != InputFiles)`. Not positive of the best way to fix this, as it's a giant wobbly mess getting GAP started. Similar special code exists in `OpenInputStream`.

**Comment by dimpase** [comment]
This does look like a bug :-) One sets `Input` to point at `InputFiles-1` and then does 
`Input->echo=1;` --- where`InputFiles` is a statically allocated array.  Why it does not overwrite anything of importance, I don't get...

**Comment by ChrisJefferson** [comment]
Nope, the `OpenInput` function reincrements it. We never write or read outside the array.

**Comment by alex-konovalov** [comment]
Closed as fixed by @ChrisJefferson in #942
